tests/tests_user_views.py

```diff
@@ -0,0 +1,124 @@
+    }
+
+
+@pytest.mark.django_db
+class TestProfileViews:
+    def test_users_detail_view_loaded_authorized(self, authorized_client, persist_user_profile):
+        users_detail_response = authorized_client.get(f'/users/{persist_user_profile.user.username}/', follow=True)
+        assert users_detail_response.status_code == 200
+        assert 'profile_details.html' in users_detail_response.templates[0].name
+
+    def test_users_detail_view_loaded_unauthorized(self, client, persist_user_profile):
+        users_detail_response = client.get(f'/users/{persist_user_profile.user.username}/', follow=True)
+        assert users_detail_response.status_code == 200
+        assert 'main/login.html' in users_detail_response.templates[0].name
+
+    def test_edit_profile_view_unauthenticated(self, client, new_profile):
+        url = reverse('users-edit-profile', kwargs={'slug': new_profile.user.username})
+        response = client.get(url, follow=True)
+        assert response.status_code == 200
+        assert 'main/login.html' in response.templates[0].name
+
+    def test_edit_profile_view_authenticated(self, client, new_profile):
+        client.login(username='testuser', password='testpassword')
+        url = reverse('users-edit-profile', kwargs={'slug': new_profile.user.username})
+        response = client.get(url, follow=True)
+        assert response.status_code == 200
+        assert 'main/login.html' in response.templates[0].name
+
+    def test_users_update_view_loaded_authorized(self, authorized_client, persist_user_profile):
+        user_detail_response = authorized_client.get(f"/users/{persist_user_profile.pk}/edit-profile/")
+        assert user_detail_response.status_code == 200
+        assert 'edit-profile.html' in user_detail_response.templates[0].name
+
+    @pytest.mark.parametrize("email, city", [(EMAIL, "city_test")])
+    def test_update_user(self, persist_user_profile, authorized_client, email, city):
+
+        profile = Profile.objects.filter(user=persist_user_profile.user).first()
+        assert profile.city != city
+        assert profile.user.email != email
+
+        new_profile = {"username": persist_user_profile.user.username,
+                       "email": email,
+                       "first_name": "",
+                       "last_name": "",
+                       "bio": "",
+                       "profession": "",
+                       "city": city,
+                       "phone_number": '0500000000',
+                       "account_type": 'B',
+                       "meeting_method": "B"}
+
+        response = authorized_client.post(f"/users/{persist_user_profile.user.username}/edit-profile/",
+                                          new_profile, follow=True)
+        profile = Profile.objects.filter(user=persist_user_profile.user).first()
+        assert response.status_code == 200
+        assert profile.city == city
+        assert profile.user.email == email
```

Comment:
I would check the content instead

Reply:
'יWhat does it mean to check the content?

Reply:
Something like [this](https://github.com/redhat-beyond/HighTeach/blob/d8a0662266c944e47d87dd52fa2275b0bb894176/tests/course/test_review_view.py#L28)

Reply:
After making posts, you get a redirection to the home page and there is nothing to check in terms of the content of the response.. Isn't it better to check in the data base that the values are indeed updated?